pkg/core/server/abci.go

```diff
@@ -255,6 +257,13 @@ func (s *Server) FinalizeBlock(ctx context.Context, req *abcitypes.FinalizeBlock
 		go s.mempl.RemoveExpiredTransactions(req.Height)
 	}
 
+	jailedValidators, err := s.handleDuplicateVoteEvidence(ctx, req)
+	if err != nil {
+		s.logger.Errorf("could not handle duplicate vote evidence: %v", err)
+	} else {
+		validatorUpdates = append(validatorUpdates, jailedValidators...)
```

Comment:
We may possibly still need to guard against duplicate validatorUpdates entries because comet vomits about that, hence why I made validatorUpdatesMap

Reply:
is the map to get rid of duplicates?